src/xcode/ENA/ENA/Source/Views/BaseElements/ENAButton.swift

```diff
@@ -80,6 +82,20 @@ class ENAButton: DynamicTypeButton {
 		highlightView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
 		addSubview(highlightView)
 
+		// MARK: - Add spinner for loading state.
+		activityIndicator = UIActivityIndicatorView(style: .medium)
+		activityIndicator.translatesAutoresizingMaskIntoConstraints = false
+		activityIndicator.isUserInteractionEnabled = false
+		addSubview(activityIndicator)
+		if let title = titleLabel {
+			activityIndicator.leadingAnchor.constraint(greaterThanOrEqualTo: leadingAnchor, constant: 8).isActive = true
+			title.leadingAnchor.constraint(equalTo: activityIndicator.trailingAnchor, constant: 8).isActive = true
+			activityIndicator.centerYAnchor.constraint(equalTo: centerYAnchor).isActive = true
+		} else {
+			activityIndicator.centerXAnchor.constraint(equalTo: centerXAnchor).isActive = true
+			activityIndicator.centerYAnchor.constraint(equalTo: centerYAnchor).isActive = true
```

Comment:
Sorry, one more thing. By default the button as a height of 50 or more, but the width can collapse.
So we need 8pt margin to left and right, please.